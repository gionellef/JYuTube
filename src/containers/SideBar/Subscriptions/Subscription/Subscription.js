import React from 'react';

import { Icon, Image, Menu } from 'semantic-ui-react';

import './Subscription.scss';


export function Subscription(props) {
  
  let live = null;
  const { broadcasting, amountNewVideos } = props;
  
  if (broadcasting) {
    live = <Icon name='signal' />
  } else if (amountNewVideos) {
    live = <span className='new-videos-count'>{ amountNewVideos }</span>
  }

  return (
    <Menu.Item>
      <div className='subscription'>
        <div>
          <Image src='http://via.placeholder.com/28x28' avatar />
          <span>{ props.label }</span>
        </div>
        { live }
      </div>
    </Menu.Item>
  );
}