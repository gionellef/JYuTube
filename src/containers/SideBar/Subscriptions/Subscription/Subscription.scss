@import '../../../../styles/shared.scss';

.subscription {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;

  i.icon {
    color: $blue;
  }

  .new-videos-count {
    color: $gray;
  }
}