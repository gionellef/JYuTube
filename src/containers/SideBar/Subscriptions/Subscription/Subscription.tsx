import * as React from 'react';

import { Icon, Image, Menu } from 'semantic-ui-react';

import './Subscription.scss';

interface Props {
  broadcasting?: boolean;
  amountNewVideos?: number;
  label: string;
}

const Subscription : React.FC<Props> = (props) => {
  let live = null;
  const { broadcasting, amountNewVideos } = props;
  
  if (broadcasting) {
    live = <Icon name='signal' />
  } else if (amountNewVideos) {
    live = <span className='new-videos-count'>{ amountNewVideos }</span>
  }

  return (
    <Menu.Item>
      <div className='subscription'>
        <div>
          <Image src='http://via.placeholder.com/28x28' avatar />
          <span>{ props.label }</span>
        </div>
        { live }
      </div>
    </Menu.Item>
  );
}

export default Subscription;