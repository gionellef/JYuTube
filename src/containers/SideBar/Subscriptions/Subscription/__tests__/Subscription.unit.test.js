import React from 'react';

import { shallow } from 'enzyme';
import { Subscription } from '../Subscription';

describe('Subscription', () => {
  test('renders empty Subscription', () => {
    const wrapper = shallow(
      <Subscription />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('renders Subscription that is recently live', () => {
    const wrapper = shallow(
      <Subscription label='exurb1a' broadcasting />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('renders Subscription with new videos and not recently live', () => {
    const wrapper = shallow(
      <Subscription label='exurb1a' amountNewVideos={15} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('renders Subscription with new videos and recently live', () => {
    const wrapper = shallow(
      <Subscription label='exurb1a' broadcasting amountNewVideos={3} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});