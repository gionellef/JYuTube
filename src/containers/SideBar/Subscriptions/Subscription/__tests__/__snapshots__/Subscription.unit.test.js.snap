// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription renders Subscription that is recently live 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        exurb1a
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription renders Subscription with new videos and not recently live 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        exurb1a
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      15
    </span>
  </div>
</MenuItem>
`;

exports[`Subscription renders Subscription with new videos and recently live 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        exurb1a
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription renders empty Subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;
